proto/cosmos/staking/query.proto

```diff
@@ -4,52 +4,81 @@ package cosmos.staking;
+		option (google.api.http).get = "/cosmos/staking/v1beta/validator/{validator_addr}";
+  }
 
   // ValidatorDelegations queries delegate info for given validator
-  rpc ValidatorDelegations (QueryValidatorDelegationsRequest) returns (QueryValidatorDelegationsResponse) {}
+  rpc ValidatorDelegations (QueryValidatorDelegationsRequest) returns (QueryValidatorDelegationsResponse) {
+		option (google.api.http).get = "/cosmos/staking/v1beta/delegations/{validator_addr}";
+  }
 
   // ValidatorUnbondingDelegations queries unbonding delegations of a validator
-  rpc ValidatorUnbondingDelegations (QueryValidatorUnbondingDelegationsRequest) returns (QueryValidatorUnbondingDelegationsResponse) {}
+  rpc ValidatorUnbondingDelegations (QueryValidatorUnbondingDelegationsRequest) returns (QueryValidatorUnbondingDelegationsResponse) {
+		option (google.api.http).get = "/cosmos/staking/v1beta/unbonding-delegations/{validator_addr}";
+  }
 
   // Delegation queries delegate info for given validator delegator pair
-  rpc Delegation (QueryDelegationRequest) returns (QueryDelegationResponse) {}
+  rpc Delegation (QueryDelegationRequest) returns (QueryDelegationResponse) {
+		option (google.api.http).get = "/cosmos/staking/v1beta/delegation/{delegator_addr}/{validator_addr}";
+  }
 
   // UnbondingDelegation queries unbonding info for give validator delegator pair
-  rpc UnbondingDelegation (QueryUnbondingDelegationRequest) returns (QueryUnbondingDelegationResponse) {}
+  rpc UnbondingDelegation (QueryUnbondingDelegationRequest) returns (QueryUnbondingDelegationResponse) {
+		option (google.api.http).get = "/cosmos/staking/v1beta/unbonding-delegation/{delegator_addr}/{validator_addr}";
+  }
 
   // DelegatorDelegations queries all delegations of a give delegator address
-  rpc DelegatorDelegations (QueryDelegatorDelegationsRequest) returns (QueryDelegatorDelegationsResponse) {}
+  rpc DelegatorDelegations (QueryDelegatorDelegationsRequest) returns (QueryDelegatorDelegationsResponse) {
+		option (google.api.http).get = "/cosmos/staking/v1beta/delegator-delegations/{delegator_addr}";
+  }
 
   // DelegatorUnbondingDelegations queries all unbonding delegations of a give delegator address
-  rpc DelegatorUnbondingDelegations (QueryDelegatorUnbondingDelegationsRequest) returns (QueryDelegatorUnbondingDelegationsResponse) {}
+  rpc DelegatorUnbondingDelegations (QueryDelegatorUnbondingDelegationsRequest) returns (QueryDelegatorUnbondingDelegationsResponse) {
+		option (google.api.http).get = "/cosmos/staking/v1beta/unbonding-delegations/{delegator_addr}";
+  }
 
   // Redelegations queries redelegations of given address
-  rpc Redelegations (QueryRedelegationsRequest) returns (QueryRedelegationsResponse) {}
+  rpc Redelegations (QueryRedelegationsRequest) returns (QueryRedelegationsResponse) {
+		option (google.api.http).get = "/cosmos/staking/v1beta/redelegations/{delegator_addr}/{src_validator_addr}/{dst_validator_addr}";
```

Comment:
Should this be `POST`? And also move `src_validator_addr` and `dst_validator_addr` to body
```suggestion
		option (google.api.http) = {
		    post: "/cosmos/staking/v1beta/delegators/{delegator_addr}/redelegations"
		    body: "..."
		};
```

Reply:
Oh may be we should stick to `get` and keep `src_validator_addr` and `dst_validator_addr` as query params?